### PR TITLE
Remove auditor-specific code

### PR DESCRIPTION
**Pull request by dav3r**
#  #

## 🗣 Description ##
This PR removes the code that created auditor groups, roles, and policies, so that this repo now creates non-admin IAM users and nothing else.

While I was updating this repo, I took the opportunity to remove the hard-coded provider name and use our now-standard method of obtaining the provider's role from a Terraform remote state instead.

## 💭 Motivation and context ##
Our goal is to make this the only repo where we provision non-admin users.  We can then have multiple repositories where those users are given permissions (e.g. [`cisagov/cool-auditor-iam`](https://github.com/cisagov/cool-auditor-iam)).

Note that this PR is directly related to [cisagov/cool-auditor-iam#1](https://github.com/cisagov/cool-auditor-iam/pull/1).

## 🧪 Testing ##
I ran a successful `terraform apply` and verified that the user accounts were untouched, while all of the auditor permissions were removed from the auditor users (I re-added those permissions a short time later while testing [cisagov/cool-auditor-iam#1](https://github.com/cisagov/cool-auditor-iam/pull/1)).

## ✅ Checklist ##

* [x] This PR has an informative and human-readable title.
* [x] Changes are limited to a single goal - _eschew scope creep!_
* [x] All relevant type-of-change labels have been added.
* [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
* [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
* [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
* [x] All new and existing tests pass.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,25 +2,16 @@
...
 [![GitHub Build Status](https://github.com/cisagov/cool-users-non-admin/workflows/build/badge.svg)](https://github.com/cisagov/cool-users-non-admin/actions)
 
-This project is used to manage IAM user accounts and group membership for
-non-admin users.  All admin users are set up in the
+This project is used to manage IAM user accounts for non-admin users.
+All admin users are set up in the
 [`cool-accounts`](https://github.com/cisagov/cool-accounts) repository
 (in the [`users` subdirectory](https://github.com/cisagov/cool-accounts/users)).
 
 ## Pre-Requisites ##
```
Comment (jsf9k) [inline README.md]: According to our Markdown style guide, this should be:
```suggestion
## Pre-requisites ##
```

**Review by jsf9k — APPROVED** [review]
I made one minor change request, but I will happily go ahead and approve.

**Review by hillaryj — APPROVED** [review]
🎂

**Review by hillaryj — APPROVED** [review]
🎂

**Review by mcdonnnj — APPROVED** [review]
🧹Tidying up!

**Pull request merged**